### DISCUSSION TITLE
Failure to find the export-set of libassert.

### DISCUSSION THREAD

**Discussion by yand-jmg** [jeremy-rifkin/libassert#108] [Ideas]
Thank you for this useful library!

In one of my other projects, I am trying to use it, following your `cmake`-based advice, in my own `cmake`-based project..

I have a library that uses libassert and links to it as per:

```
target_link_libraries(libfoo_archive
	PUBLIC
		libassert::assert
)
```
`libfoo` is later exported:
```
    install(
        TARGETS libfoo
        EXPORT libfooTargets
        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT runtime
        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT development
        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBEXECDIR}	COMPONENT runtime
    )
```
Which cause `cmake` to emit this error:
```
CMake Error in CMakeLists.txt:
  export called with target "libfoo" which requires target "libassert-lib"
  that is not in any export set.
```
Can you advise regarding how I might fix this problem, please?

**Comment by jeremy-rifkin** [comment]
Hi thanks for opening this. I had this issue recently in the upstream library (cpptrace) and now that it's fixed there I can fix it here too.

**Reply by jeremy-rifkin** [reply]
Now fixed on dev, I'll do a patch release with this and some other fixes some time soon

**Reply by yand-jmg** [reply]
Thanks a lot - let me know and might you put a link into the pull request, so I can ensure my library is "done correctly" in cmake, in case I have made an error there!

**Reply by yand-jmg** [reply]
This has been resolved in v2.1.4, also see resolved issue: #113 .
